### PR TITLE
[Core/Studio] Added support for user-installed fonts on Windows 10 (1809)

### PR DESCRIPTION
**Pull request by ice0**
Fix https://github.com/synfig/synfig/issues/1873

### CODE REVIEW

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp [hunk 1]
```diff
@@ -675,7 +684,18 @@ Layer_Freetype::new_face(const String &newfont)
...
 
 #ifdef _WIN32
-	possible_font_directories.push_back("C:\\WINDOWS\\FONTS\\");
+	// All users fonts
+	std::string windir = Glib::getenv("windir");
+	if (windir.empty()) {
+		possible_font_directories.emplace_back("C:\\WINDOWS\\FONTS\\");
+	} else {
+		possible_font_directories.emplace_back(windir + "\\Fonts\\");
+	}
```
Comment (rodolforg) [inline synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:693]: What about:
```c++
	if (windir.empty())
		windir = "C:\\WINDOWS;
	possible_font_directories.emplace_back(windir + "\\Fonts\\");

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp [hunk 2]
```diff
@@ -675,7 +684,18 @@ Layer_Freetype::new_face(const String &newfont)
...
+	// Windows 10 (1809) Added local user fonts installed to C:\Users\%USERNAME%\AppData\Local\Microsoft\Windows\Fonts
+	std::string localdir = Glib::getenv("LOCALAPPDATA");
+	if (!localdir.empty()) {
+		possible_font_directories.emplace_back(localdir + "\\Microsoft\\Windows\\Fonts\\");
+	}
```
Comment (rodolforg) [inline synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:698]: As the code is duplicated, would it worth to create a simple static (='local', not in the sense of a class method) function near file header?
```c++
#ifdef _WIN32
// Windows 10 (1809) Added local user fonts installed to C:\Users\%USERNAME%\AppData\Local\Microsoft\Windows\Fonts
static std::string get_user_font_folder() {
...
}
#endif 
```

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp (hunk 1 above)
Comment (ice0) [reply synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:693]: As for me, it's just faster, but I can change it if it's look more clear

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp (hunk 2 above)
Comment (ice0) [reply synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:698]: Any idea where to put it? As I can see where is no shared headers between `lyr_freetype` and `widget_fontfamily`.

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:693]: It's a matter of taste, actually. But I think it's error prone.

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp (hunk 2 above)
Comment (rodolforg) [reply synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:698]: I don't know.
I think Synfig library should provide those system and resource paths :P 
Anyway, not important.

**Comment by ice0** [comment]
Can't backport to 1.4 because it doesn't have required changes to text layer

**Pull request merged**